wordpress-mobile/WordPress-Android
# Reader move tab layout to parent part 5 - restore lastly selected tab

Partially fixes #11849

Merge instructions:
1. Review this PR
1. Make sure https://github.com/wordpress-mobile/WordPress-Android/pull/11872 is merged
2. Change the target branch to `master-feature-branch/move-reader-tablayout-to-parent`
3. Remove the "Not Ready for merge" label
3. Merge this PR

This is a 5th PR in a series of PRs which is trying to introduce a regular TabLayout+ViewPager2 in the main Reader view. The current implementation uses FilteredRecyclerView component that has some downsides (eg. swipe gesture doesn't work). These changes will also help us implement the improvements in the ReaderImprovements project. 
P.S. The known issues/remaining tasks are listed in #11849 

Goal of this PR is to (re)store lastly selected tab in the Reader.

To test:
1. Select "Liked" tab in the Reader
2. Kill the app
3. Open the app and notice the "Liked" tab is selected
4. Repeat the steps for "Discover"

PR submission checklist:
